### PR TITLE
Reset/Forgot Password - Back End

### PR DESCRIPTION
**Pull request by danielgnz**
I have created the basic functionality for both Reset and ForgotPassword.js: (See: [Task](https://trello.com/c/01lN84sF/163-reset-password-resetpasswordjs-forgotpasswordjs))

This implementation goes hand in hand with: https://github.com/zero-to-mastery/mappypals/pull/115

What I have done:

- Checked if user's email input is in the database
- Generated unique token which is available 15 minutes
- Created a gmail account  (check env example for credentials) and used it to send the reset link url to the client
- Updated the database with the client's newly set password

Should you have any concerns/comments you would like to make, please let me know. I will make any changes needed.

### CODE REVIEW

File: env.example [hunk 1]
```diff
@@ -1,2 +1,4 @@
 MONGO_URI="mongodb+srv://mappypals:ZozaDZQ7FDKmZZIq@mappypals-hzxcd.mongodb.net/test?retryWrites=true"
-JWT_SECRET="w23Q$8$7sK$aQod83YQv2#29vOV2#0V"
\ No newline at end of file
+JWT_SECRET="w23Q$8$7sK$aQod83YQv2#29vOV2#0V"
+EMAIL_ADDRESS="team.mappypals@gmail.com"
+EMAIL_PASSWORD="mappypalsadmin"
```
Comment (ArnasDickus) [inline env.example:4]: The file is not formatted try running `npm run dev:format`

File: server.js [hunk 2]
```diff
@@ -24,6 +26,8 @@ mongoose
     .connect(db, { useNewUrlParser: true })
     .then(() => {
         console.log('MongoDB Connected');
+        // User.findOne({ email: 'danibtechnology@gmail.com'})
+        // 	.then(user => console.log(user));
```
Comment (ArnasDickus) [inline server.js]: Not sure if comment here is necessary.

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
Hello, thanks for adding PR
Since I already tested [front-end](https://github.com/zero-to-mastery/mappypals/pull/115) with back-end most functionality works.
So I didn't find problems here. Apart from the formating. I'm going to approve this when you fix changes for front-end PR

File: server.js (hunk 2 above)
Comment (danielgnz) [reply server.js]: I used that for debugging. It's not necessary, indeed.

File: env.example (hunk 1 above)
Comment (danielgnz) [reply env.example:4]: I have formatted it now.

**Comment by ArnasDickus** [comment]
@danielgnz  Alright, since you fixed all issues I had. Time to approve it.

**Review by ArnasDickus — APPROVED** [review]

**Pull request merged**